biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java

```diff
@@ -670,18 +670,18 @@ public String toString() {
 		return runs;
 	}
 
-	Map<File, Resource> doExports(Map<File, List<Attrs>> entries) {
+	Map<File, Resource> doExports(Map<File, List<Attrs>> entries) throws Exception {
 		Map<File, Resource> result = new LinkedHashMap<>();
 
 		for (Entry<File, List<Attrs>> e : entries.entrySet()) {
 
 			for (Attrs attrs : e.getValue()) {
 
 				File file = e.getKey();
+				Run run = null;
 				try {
 
-					Run run = Run.createRun(getProject().getWorkspace(), file);
-
+					run = Run.createRun(getProject().getWorkspace(), file);
```

Comment:
Run is a Closeable, to use a try-with-resources statement here to close the Run.

```java
try (Run run = Run.createRun(getProject().getWorkspace(), file)) {
  ...
}
```